packages/nextjs/src/server/utils/withClerkMiddleware.ts

```diff
@@ -78,14 +78,15 @@ export function withClerkMiddleware(handler: NextMiddleware, opts: WithAuthOptio
       throw 'withClerkMiddleware: Auth result could not be determined';
     }
 
-    setAuthResultOnRequest({ req, authResult });
+    // Set auth result on request in a private property so that middleware can read it too
+    setPrivateAuthResultOnRequest({ req, authResult });
```

Comment:
❓ Does this also mean that AuthResult is set for successful cases as well? If yes, it's a good improvement to have it omnipresent.